### PR TITLE
Spacing tokens update

### PR DESCRIPTION
**Pull request by ameseee**
#### What does this PR do?

- Updates spacing tokens to use a 0, 1, 2, 3, etc. systems rather than the 0, 2, 4, 8, 16 it was previously using. 
- Refactored the spacing tokens to utilize an @each and make file shorter, less repetition

#### Where should the reviewer start?

- `tokens/spacing.scss`

#### Any background context you want to provide?

- Spacing tokens were being used by several utils and in doc site styling itself, so I went back through and changed each to the use what is now the correct variable. It made me nervous but I believe I got them all.

#### What are the relevant tickets?

https://3.basecamp.com/3494409/buckets/19192671/todos/3455119332 

#### Screenshots (if appropriate)

Screen Shot 2021-05-13 at 10 30 56 AM

### CODE REVIEW

File: css/design_system/index.scss [hunk 1]
```diff
@@ -8,6 +8,7 @@
 @import "utils/spacing";
 @import "utils/borders";
 @import "utils/text";
+@import "utils/shadows";
```
Comment (tannerwelsh) [inline css/design_system/index.scss:11]: Looks like the shadows changes got accidentally included in this PR. Should we come back to this PR after shadows have been merged in?

File: css/design_system/tokens/spacing.scss [hunk 2]
```diff
@@ -1,114 +1,29 @@
...
+  "10": 2.5rem,
+  "11": 3rem,
+  "12": 3.5rem,
+  "13": 4rem
+);
+
 :root {
-  /**
-   * @title --s-space-0
+  @each $name, $size in $spacing-options {
```
Comment (tannerwelsh) [inline css/design_system/tokens/spacing.scss:19]: Soooo much better

**Review by tannerwelsh — APPROVED** [review]
Thanks for painstakingly updating all the sizing variables! I think it should be a lot more usable going forward, no more needing to remember what the next interval is.

File: css/design_system/index.scss (hunk 1 above)
Comment (ameseee) [reply css/design_system/index.scss:11]: YUP! And I have a question for you next time we meet about this type of "accident" RE my workflow

**Pull request merged**